UnitsNet.Tests/CustomCode/MassConcentrationTests.cs

```diff
@@ -0,0 +1,159 @@
+        protected override double CentigramsPerMilliliterInOneKilogramPerCubicMeter => 1e-1;
+        protected override double GramsPerDeciliterInOneKilogramPerCubicMeter => 1e-1;
+        protected override double DecigramsPerMilliliterInOneKilogramPerCubicMeter => 1e-2;
+        protected override double KilogramsPerLiterInOneKilogramPerCubicMeter => 1e-3;
+        protected override double GramsPerMilliliterInOneKilogramPerCubicMeter => 1e-3;
+        protected override double GramsPerCubicCentimeterInOneKilogramPerCubicMeter => 1e-3;
+        protected override double TonnesPerCubicMeterInOneKilogramPerCubicMeter => 1e-3;
+        protected override double GramsPerCubicMillimeterInOneKilogramPerCubicMeter => 1e-6;
+        protected override double KilogramsPerCubicCentimeterInOneKilogramPerCubicMeter => 1e-6;
+        protected override double KilogramsPerCubicMillimeterInOneKilogramPerCubicMeter => 1e-9;
+        protected override double TonnesPerCubicCentimeterInOneKilogramPerCubicMeter => 1e-9;
+        protected override double TonnesPerCubicMillimeterInOneKilogramPerCubicMeter => 1e-12;
+
+        protected override double PoundsPerImperialGallonInOneKilogramPerCubicMeter => 1.002241e-2;
+        protected override double KilopoundsPerCubicFootInOneKilogramPerCubicMeter => 6.242796e-5;
+        protected override double KilopoundsPerCubicInchInOneKilogramPerCubicMeter => 3.6127292e-8;
+        protected override double PoundsPerCubicFootInOneKilogramPerCubicMeter => 6.242796e-2;
+        protected override double PoundsPerCubicInchInOneKilogramPerCubicMeter => 3.61272923e-5;
+        protected override double PoundsPerUSGallonInOneKilogramPerCubicMeter => 8.3454045e-3;
+        protected override double SlugsPerCubicFootInOneKilogramPerCubicMeter => 0.00194032;
+        #endregion
+
+        [Theory]
+        [InlineData(2, MassConcentrationUnit.KilogramPerCubicMeter, 3, VolumeUnit.CubicMeter, 6, MassUnit.Kilogram)]
+        public static void MassConcentrationTimesVolumeEqualsMass(
+            double massConcValue, MassConcentrationUnit massConcUnit,
+            double volumeValue, VolumeUnit volumeUnit,
+            double expectedMassValue, MassUnit expectedMassUnit, double tolerance= 1e-5)
+        {
+            var massConcentration = new MassConcentration(massConcValue, massConcUnit);
+            var volume = new Volume(volumeValue, volumeUnit);
+
+            Mass mass = massConcentration * volume;
+            AssertEx.EqualTolerance(mass.As(expectedMassUnit), expectedMassValue, tolerance);
+        }
+
+        [Fact]
+        public static void VolumeTimesMassConcentrationEqualsMass()
+        {
+            Mass mass = Volume.FromCubicMeters(3) * MassConcentration.FromKilogramsPerCubicMeter(2);
+            Assert.Equal(mass, Mass.FromKilograms(6));
+        }
+
+        [Fact]
+        public void ExpectMolarityConvertedToMassConcentrationCorrectly()
```

Comment:
If this is testing `Molarity.ToMassConcentration()` then this test should be moved to `MolarityTests` and renamed slightly. "Correctly" is a bit vague, if feasible explain exactly what is the expected outcome.

Reply:
Hmm, this one (along with another from VolumeConcentration) were actually duplicated in both source and target tests. I'm going to leave XX  = input.ToXX(..) in the TestsXX. Although I'm not exactly sure where to define the actual operator- in XX or in input.class :)

Reply:
Now I even understand why they got duplicated- my first intuition was that I'd place them where the correct TolerenceXX value lies. Then I decided to test "the operator" class and there I think there must have been a brain crash :)

Reply:
If you are testing a method, then I think the test belongs to that class.
If you are testing arithmetic operators, then if I recall correctly I think we have tried to gather those on the first/left-hand argument type. 

Example: 
```c#
// put this implementation in Force.extra.cs and the test in ForceTests
Torque t = Force.Zero * Length.Zero;
```

Reply:
Following the Method/FirstOperator : TestClass mapping, I've shuffled some of the tests around. 
Also moved the Mass/MolarMass operator to the Mass.extra class (removing the MolarMass.extra)

Reply:
Nice